### PR TITLE
Execute redshift import on schedule

### PR DESCRIPTION
**Pull request by sureshc**
Scheduling export from production Aurora MySQL to Redshift to execute once a day at 2AM UTC (6PM PST / 7PM PDT).  It typically completes in 10 hours.

### Background

https://codedotorg.atlassian.net/browse/INF-242

## Testing story

Several iterations of this job have been successfully executed manually on `production-daemon` several times.  The Data stack changes need to be applied before this job executes for the first time after being deployed.

Validation of the Data stack changes:
```
bundle exec rake stack:data:validate RAILS_ENV=production

Listing changes to existing stack `DATA-production`:
Add AuroraCloneEndpoint [AWS::DMS::Endpoint]
Remove AuroraReaderEndpoint [AWS::DMS::Endpoint]
Modify DMSCronLevelSourcesPii [AWS::DMS::ReplicationTask] Properties Replacement: Conditional (SourceEndpointArn, SourceEndpointArn)
Modify DMSCronPii [AWS::DMS::ReplicationTask] Properties Replacement: Conditional (SourceEndpointArn, SourceEndpointArn)
Modify DMSCronUserHierarchyPii [AWS::DMS::ReplicationTask] Properties Replacement: Conditional (SourceEndpointArn, SourceEndpointArn)
Modify DMSCronUserHierarchy [AWS::DMS::ReplicationTask] Properties Replacement: Conditional (SourceEndpointArn, SourceEndpointArn)
Modify DMSCronUserLevels [AWS::DMS::ReplicationTask] Properties Replacement: Conditional (SourceEndpointArn, SourceEndpointArn)
Modify DMSCron [AWS::DMS::ReplicationTask] Properties Replacement: Conditional (SourceEndpointArn, SourceEndpointArn)
```

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: bin/cron/export_mysql_database_to_redshift [hunk 1]
```diff
@@ -6,39 +6,64 @@ require_relative '../../dashboard/config/environment'
...
-
   # Spawn one thread for each replication task, so we can start and monitor them independently.
-  threads = []
-  Thread.abort_on_exception = true
+  daily_threads = []
   daily_tasks.each do |task|
-    # Wait 16 hours (make 96 attempts, once every 10 minutes) for tasks to complete successfully.
-    # The user_levels task takes the longest (~10 hours).
-    threads << Thread.new {task.start(TASK_EXECUTION_TIME.div(TASK_STATUS_DELAY), TASK_STATUS_DELAY)}
+    daily_threads << Thread.new {task.start(16.hours.div(TASK_STATUS_DELAY), TASK_STATUS_DELAY)}
```
Comment (wjordan) [inline bin/cron/export_mysql_database_to_redshift]: Just checking- did we remove the `TASK_EXECUTION_TIME` constant and comment just to make this line of code more compact?

File: bin/cron/export_mysql_database_to_redshift [hunk 2]
```diff
@@ -6,39 +6,64 @@ require_relative '../../dashboard/config/environment'
...
   end
-  threads.each(&:join)
+  daily_threads.each(&:join)
```
Comment (wjordan) [inline bin/cron/export_mysql_database_to_redshift]: Not necessary to change (just my preference for comparison), here's a more functional style without any local-vars:
```ruby
Cdo::DMS::ReplicationTask.tasks_by_frequency('daily').map do |task|
  Thread.new {task.start(16.hours.div(TASK_STATUS_DELAY), TASK_STATUS_DELAY)}
end.each(&:join)
```

**Review by wjordan — APPROVED** [review]
I think this looks great! The added comments make it very easy to follow what's going on in this fairly complicated script, very much appreciated.

File: bin/cron/export_mysql_database_to_redshift (hunk 1 above)
Comment (sureshc) [reply bin/cron/export_mysql_database_to_redshift]: The task execution time is different for daily tasks than weekly tasks, so I removed the constant and passed in the different time for each scenario.  Would declaring two different constants help( `DAILY_TASK_EXECUTION_TIME` and `WEEKLY_TASK_EXECUTION_TIME`)?

File: bin/cron/export_mysql_database_to_redshift (hunk 2 above)
Comment (sureshc) [reply bin/cron/export_mysql_database_to_redshift]: That's much better.   Done!

File: bin/cron/export_mysql_database_to_redshift (hunk 1 above)
Comment (wjordan) [reply bin/cron/export_mysql_database_to_redshift]: sure that works- I think the extra comments are helpful! (I didn't even notice the separate `40.hours` constant)

**Pull request merged**